tools/wptrunner/wptrunner/executors/protocol.py

```diff
@@ -132,21 +132,6 @@ def wait(self):
         """Wait indefinitely for the browser to close"""
         pass
 
-    @property
-    def current_window(self):
-        """Return a handle identifying the current top level browsing context
-
-        :returns: A protocol-specific handle"""
-        pass
-
-    @abstractmethod
-    def set_window(self, handle):
```

Comment:
I don't think this shuld be removed; it looks like most executors need to implement it and those that don't can explictly opt out.

Reply:
How about moving it to a new subclass of `BaseProtocolPart`? Usually a class name is evident when an extension like this is necessary, but the best I can think of is `MutableWindowProtocolPart`. Not great.

Reply:
Subclasses don't necessarily need this method, so declaring an `abstractmethod` (i.e. technically requiring subclasses to implement) seems contradictory to me. Other contributors (and me in 3 months) aren't likely to understand this. They might implement the method in some other subclass (contributing dead code). They also might invoke the method from outside the subclass, generically (relying on a method with partial availability).

Adding a new subclass in the inheritance chain would allow us to declare the interface in an object-oriented way without inaccurately representing the true requirements. It was just a guess that the OO pattern is what's important here.

A different compromise would be to keep the `abstractmethod` in the base class as per your request and document the optional nature of the required method. I could use a hand writing the code comment, though, because I don't understand why it's important to declare the method in this way.